File: app/src/main/java/com/kenjigames/ividsmusic/domain/model/Album.kt
package com.kenjigames.ividsmusic.domain.model

/**
 * Data class representing a music album in the domain layer.
 * Extends [MusicItem] for album card rendering.
 *
 * @property id Unique album identifier
 * @property title Name of the album
 * @property artistName Name of the performing artist
 * @property artist Typed reference to performing [Artist] entity
 * @property coverUrl High-resolution album cover artwork URL
 * @property trackCount Total number of songs in the album
 * @property releaseDate Release year or date string
 * @property songs List of tracks included in the album
 */
data class Album(
    override val id: String,
    val title: String,
    val artistName: String,
    val artist: Artist? = null,
    override val coverUrl: String,
    val trackCount: Int = 0,
    val releaseDate: String = "",
    val songs: List<Song> = emptyList()
) : MusicItem() {
    override val displayTitle: String get() = title
}

File: app/src/main/assets/logic/domain/Album.js
/**
 * Domain entity representing a music album.
 */
export class Album {
    #id = '';
    #title = '';
    #artist = '';
    #artistId = null;
    #cover = '';
    #trackCount = 0;
    #releaseYear = '';

    /**
     * Creates a new Album instance.
     * @param {Object} data - Album raw properties
     */
    constructor(data = {}) {
        this.#id = String(data.id || '');
        this.#title = data.title || 'Unknown Album';
        this.#artist = data.artist || 'Unknown Artist';
        this.#artistId = data.artistId || null;
        this.#cover = data.cover || 'svg/album-placeholder.svg';
        this.#trackCount = Number(data.trackCount) || 0;
        this.#releaseYear = String(data.releaseYear || '');
    }

    /** Gets album unique identifier */
    get id() { return this.#id; }

    /** Gets album title */
    get title() { return this.#title; }

    /** Gets primary artist name */
    get artist() { return this.#artist; }

    /** Gets primary artist ID */
    get artistId() { return this.#artistId; }

    /** Gets cover artwork URL */
    get cover() { return this.#cover; }

    /** Gets total track count */
    get trackCount() { return this.#trackCount; }

    /** Gets album release year */
    get releaseYear() { return this.#releaseYear; }

    /** Gets item type identifier */
    get type() { return 'album'; }

    /**
     * Converts domain model to plain JSON object representation.
     * @returns {Object} Plain object
     */
    toJSON() {
        return {
            type: 'album',
            id: this.#id,
            title: this.#title,
            artist: this.#artist,
            artistId: this.#artistId,
            cover: this.#cover,
            trackCount: this.#trackCount,
            releaseYear: this.#releaseYear
        };
    }
}
